From 46f04f3808e8b8c9608d0974c4cede1e78110b7f Mon Sep 17 00:00:00 2001
From: Daniel Golle <daniel@makrotopia.org>
Date: Mon, 25 Jul 2022 04:17:44 +0200
Subject: [PATCH 1/1] devices: add MediaTek MT7986 WiSoC

Add detection of the MediaTek MT7986 WiSoC using the compatible string
from device tree.

Signed-off-by: Daniel Golle <daniel@makrotopia.org>
---
 devices.txt      | 1 +
 iwinfo_nl80211.c | 6 ++++++
 2 files changed, 7 insertions(+)

--- a/hardware.txt
+++ b/hardware.txt
@@ -196,6 +196,7 @@
 0x14c3 0x7650 0x14c3 0x7650    0      0  "MediaTek" "MT7610E"
 0x14c3 0x7662 0x14c3 0x7662    0      0  "MediaTek" "MT76x2E"
 0x14c3 0x7915 0x14c3 0x7915    0      0  "MediaTek" "MT7915E"
+0x14c3 0x7986 0x14c3 0x7986    0      0  "MediaTek" "MT7986"
 0x14e4 0xaa52 0x14e4 0xaa52    0      0  "Broadcom" "BCM43602"
 0x02d0 0xa9a6 0x0000 0x0000    0      0  "Cypress"  "CYW43455"
 0x1ae9 0x0310 0x1ae9 0x0000    0      0  "Wilocity" "Wil6210"
--- a/iwinfo_nl80211.c
+++ b/iwinfo_nl80211.c
@@ -3410,7 +3410,13 @@ static int nl80211_hardware_id_from_fdt(struct iwinfo_hardware_id *id, const cha
 		id->device_id = 0x7622;
 		id->subsystem_vendor_id = 0x14c3;
 		id->subsystem_device_id = 0x7622;
+	} else if (!strcmp(compat, "mediatek,mt7986-wmac")) {
+		id->vendor_id = 0x14c3;
+		id->device_id = 0x7986;
+		id->subsystem_vendor_id = 0x14c3;
+		id->subsystem_device_id = 0x7986;
 	}
+
 	return (id->vendor_id && id->device_id) ? 0 : -1;
 }
 
-- 
2.30.2
